### ISSUE TITLE
CommentsIndentationCheck doesn't report errors at end of block

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#2078]
Code:
````
package test;

public class TestClass3 {
	public void test() {
		int a = 0;
			// wrong <-- line 6
	}
	
		// wrong <-- line 9
}
````
Config:
````
<module name="Checker">
	<property name="charset" value="ASCII" />
	<property name="severity" value="error" />
	<property name="fileExtensions" value="java" />
	
	<module name="TreeWalker">
		<module name="CommentsIndentation">
		</module>
	</module>
</module>
````
Output:
````
Starting audit...
Audit done.
````

Line 6 and 9 are not reported because checkstyle is only looking at the next statement, and when there is no next statement it drops the comment.

**Comment by romani** [comment]
Can you share with chunk of real code where people put comment at the if the block ? I need to see such a weird code and what content their consider useful at that place

**Comment by rnveach** [comment]
These are real examples from my code, just simplified.

Example 1: I like to use a single line border to separate a group of methods in my code. 
````
class MyClass {
    // //////////////////////////////////////////////////////////////////////////////

    public void t() {}

    // //////////////////////////////////////////////////////////////////////////////
}
````

Example 2: TODOs that are still missing
````
    public void t() {
        work();
        // TODO: missing functionality
    }
````

Example 3: Commented out code
````
    public void t() {
        work();
        // refreshDisplay();
    }
````

Example 4: Previous line too long for comment
````
    public void t() {
        wooooooooooooooooooooooooooork();
       // ^-- some hint
    }
````

Example 5: Empty catch area with reason that EmptyCatchBlockCheck requires
````
    public void t() {
        try {
                ...
        } catch (Exception e) {
                // ignore exception, reason
        }
    }
````

**Comment by romani** [comment]
Thanks a lot!

**Comment by romani** [comment]
@MEZk , please take a look.

**Comment by MEZk** [comment]
@romani @ rnveach 
In current implementation CommentsIndentationCheck does not rise violations when comment is splaced immediately before ```}``` (this is checked directly in check's logic). For example,

````java
public void test() {
    // some comment
    int a = 0;
     // comment
}
````

If we change check's logic, the following example will rise violation that it should be formatted like this:

````java
public void test() {
    // some comment
    int a = 0;
// comment
}
````

Did I understand correctly?

**Comment by rnveach** [comment]
@MEZk
> Did I understand correctly?

From what i have seen the formatter do, it puts all direct childs of the block at the same column position. In this case, it won't be same column position as the '}', but one indent over like the line above it.

Example of properly formatted comments using Eclipse:
```
public void test() {
    // some comment
    int a = 0;
    // comment
}
public void test2() {
    // alone comment
}
public void test3() {
    try {
    } catch(Exception e) {
        // alone comment
    }
}
public void test4() {
    if () {
    }
    // alone comment
}
```

**Comment by romani** [comment]
> If we change check's logic, the above example will rise violation that it should be formatted like this. Did I understand correctly?

no. last comment is still belong to scope of method, and all its content should be indented to the right, comment should be treated as any other statement of that method.

**Comment by romani** [comment]
merged
